Assets/Scripts/UI/StatBar.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class StatBar : MonoBehaviour
+{
+    public string SpriteLocation = "Sprites/PlayerHungerStates_Sprite";
+    
+    private int _statusValue = 100;
+
+    private Sprite[] _barStates;
+
+    private Image[] _images;
+
+    private Animator _animator;
+    
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+        _barStates = Resources.LoadAll<Sprite>(SpriteLocation);
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+         
+    }
+
+    public void UpdateBar(int value)
+    {
+        var newValue = value;
+        var oldValue = _statusValue;
+
+        if (newValue > 100 || newValue < 0)
+        {
+            Debug.Log("Status bar value has to be between 0 and 100");
+        }
+        else if (newValue == oldValue)
+        {
+            Debug.Log("new status bar value has to differ from the old value");
```

Comment:
That shouldn't be an error. This would allow us to animate the bar, e.g. highlight it, even though the player had full hunger/thirst

Reply:
removed